.github/workflows/npmpublish.yml

```diff
@@ -0,0 +1,34 @@
+name: "NPM Publish and GitHub tag"
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+          registry-url: https://registry.npmjs.org/
+      - run: yarn install
+      - run: npm publish --access public
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
+      # Set PACKAGE_VERSION, pull from package.json
+      - name: package-version
+        run: node -p -e '`PACKAGE_VERSION=${require("./package.json").version}`' >> $GITHUB_ENV
+      # Create the tag using the new PACKAGE_VERSION value
+      - name: Create tag
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ github.token }}
```

Comment:
Should be `secrets.GITHUB_TOKEN`?

Reply:
Hey, my [tag](https://github.com/jay-hodgson/Synapse-React-Client/tags) was created when I tested this on my forked repo (pulling into my main), so it seems to work.  But is secrets.GITHUB_TOKEN preferable???